mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -638,7 +677,35 @@ LogicalResult tosa::PadOp::inferReturnTypeComponents(
                           paddingValues[i * 2 + 1]);
   }
 
-  inferredReturnShapes.push_back(ShapedTypeComponents(outputShape));
+  inferredReturnShapes.push_back(ShapedTypeComponents(outputShape, inputType));
+  return success();
+}
+
+LogicalResult PadOp::verify() {
+  ShapedType inputType = llvm::cast<ShapedType>(getInput1().getType());
```

Comment:
I'm not sure whether we can assume here that the type is a ShapedType. Could you please check whether the this function will be invoked when the type is different in IR? If yes, please check with dyn_cast.

Reply:
I am not sure how to give an input that is not shapedType. Can you give an example.

Reply:
Sure! You could use the element types instead of tensor types, like:
`%0 = "tosa.concat"(%arg0, %arg1) {axis = 1 : i64} : (f32, i8) ->f32`
to have non shapedtypes on the operands/results.

Reply:
I tested with inputs that are not tensor types and I got an expected error as below
`op operand #0 must be ranked tensor of number values, but got 'f32'`

I think this is because of the padOp definition in TosaOps.td where input type is specified as a `Tosa_RankedTensor`.

I added a new test case for this in invalid.mlir and also added a new check in verify method to make sure the input rank is not 0 as per the spec (https://www.mlplatform.org/tosa/tosa_spec.html#_pad)